### PR TITLE
fix(user-onboarding): add properties to tourState, save state initially, add tourComplete observable

### PR DESCRIPTION
**Pull request by barleendhaliwal**
## Description
added status and prop to tourState model, fix state not getting saved in case undefined, add tourComplete observable that can be subscribed to, add setSessionIdGenerator function in tourStoreService

Fixes # gh-0

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/user-onboarding/projects/user-onboarding-lib/src/models/index.ts [hunk 1]
```diff
@@ -6,3 +6,5 @@ export * from './TourStep';
 export * from './parameterModel';
 export * from './videoModel';
 export * from './filterFunction';
+export * from './Props';
```
Comment (samarpan-b) [inline packages/user-onboarding/projects/user-onboarding-lib/src/models/index.ts:9]: follow file naming convention. File names cant start with caps

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts [hunk 2]
```diff
@@ -44,6 +44,21 @@ export class TourStoreServiceService {
   private readonly defaultLTCommand = new LoadTCommand(this.storage);
   private readonly defaultDSCommand = new DeleteSCommand(this.storage);
   private readonly defaultDTCommand = new DeleteTCommand(this.storage);
+  private sessionGenerator: () => string = () => {
```
Comment (samarpan-b) [inline packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts:47]: cant we use any existing uuid library ?

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: packages/user-onboarding/projects/user-onboarding-lib/src/models/index.ts (hunk 1 above)
Comment (barleendhaliwal) [reply packages/user-onboarding/projects/user-onboarding-lib/src/models/index.ts]: fixed naming convention for all files in models

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts (hunk 2 above)
Comment (barleendhaliwal) [reply packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts:48]: fixed

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts [hunk 3]
```diff
@@ -153,23 +190,31 @@ export class TourServiceService {
...
             };
             const wrapperNormalNext = () => {
-              this.tourStoreService.saveState({
-                tourId: tourInstance.tourId,
-                state: {
-                  sessionId: this.tourStoreService.getSessionId(),
-                  step: e.nextStepId,
-                },
-              });
+              this.tourStoreService
```
Comment (akshatdubeysf) [inline packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts:193]: Can make this a function, and use that everywhere

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts [hunk 4]
```diff
@@ -298,11 +364,18 @@ export class TourServiceService {
               scrollTo: {behavior: 'smooth', block: 'center'},
             },
           });
+          this.tour.on(
+            'complete',
+            (event: {index: number; tour: Shepherd.Tour; tourId: string}) => {
```
Comment (akshatdubeysf) [inline packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts:369]: Create a type for this instance.

**Review by akshatdubeysf — APPROVED** [review]

**Review by samarpan-b — COMMENTED** [review]
File names are still incorrect @barleendhaliwal . They should be kebab case. I am merging it for now. Please fix the pending comments in a new PR separately.

**Comment by samarpan-b** [comment]
Also fix sonar in next PR. @barleendhaliwal

**Pull request merged**